pkg/oc/cli/cmd/rollout/cancel.go

```diff
@@ -98,27 +116,38 @@ func (o *CancelOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, out io.
 		return err
 	}
 
-	r := f.NewBuilder().
+	o.Printer = func(successMsg string) (printers.ResourcePrinter, error) {
+		o.PrintFlags.Complete(successMsg)
+		return o.PrintFlags.ToPrinter()
+	}
+
+	o.Builder = f.NewBuilder()
+	o.Resources = args
+	return err
+}
+
+func (o CancelOptions) Run() error {
+	r := o.Builder.
 		WithScheme(ocscheme.ReadingInternalScheme).
-		NamespaceParam(cmdNamespace).DefaultNamespace().
-		FilenameParam(enforceNamespace, &o.FilenameOptions).
-		ResourceTypeOrNameArgs(true, args...).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		FilenameParam(o.NamespaceRequired, &o.FilenameOptions).
+		ResourceTypeOrNameArgs(true, o.Resources...).
 		ContinueOnError().
 		Latest().
 		Flatten().
 		Do()
-	err = r.Err()
+	err := r.Err()
 	if err != nil {
 		return err
 	}
 
-	o.Infos, err = r.Infos()
-	return err
-}
+	infos, err := r.Infos()
+	if err != nil {
+		return err
+	}
 
-func (o CancelOptions) Run() error {
 	allErrs := []error{}
-	for _, info := range o.Infos {
+	for _, info := range infos {
 		config, ok := info.Object.(*appsapi.DeploymentConfig)
```

Comment:
Are you sure you're working with internal here?

Reply:
Thanks, updated builder Scheme param, and this line to use appsapiv1